app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,31 @@
+class NewsletterFinder
+  include Enumerable
+
+  delegate :each, to: :newsletters_list
+
+  def initialize(type)
+    @type = type
+  end
+
+  private
+
+  def newsletters_list
+    @newsletters_list ||= newsletters_with_file_extension.map do |newsletter|
+      newsletter.gsub(/\.pdf/, "")
+    end
+  end
+
+  def newsletters_with_file_extension
+    unsorted_newsletters.sort do |date_1, date_2|
+      Date.parse("1-#{date_2}") <=> Date.parse("1-#{date_1}")
```

Comment:
@halogenandtoast It isn't a problem. The one is just representing the day. I already have two flyers for January and two for December up already. I didn't think of the day when I named the files as I only associate them with the month. I could change the file names to include the 1 so this can be removed but is it worth it?